pandas/tests/indexes/datetimes/test_ops.py

```diff
@@ -175,6 +175,17 @@ def test_round(self):
             tm.assertRaisesRegexp(ValueError, msg, rng.round, freq='M')
             tm.assertRaisesRegexp(ValueError, msg, elt.round, freq='M')
 
+            # GH 14440
```

Comment:
So the rounding is happening correctly for microseconds but not with nanoseconds:

```
In [7]: pd.DatetimeIndex(['2016-10-17 12:00:00.0015']).round('ns')
Out[7]: DatetimeIndex(['2016-10-17 12:00:00.001499904'], dtype='datetime64[ns]', freq=None)
```

The rounding methodology seems sound. I am unsure if this is a limitation of the date going from int64 to float64 to int64 as this is essentially what is happening:

```
(Pdb) np.round(np.array([1476705600001500000])/1.).astype('i8')
array([1476705600001499904])
```